specifications/sep-payment-channel-mechanism.md

```diff
@@ -0,0 +1,631 @@
+
+### Computed Values
+
+The two participants frequently use the following computed values:
+
+- s_i, the _iteration sequence number_, is the sequence number that iteration
+i's transaction set starts at. Assuming the history of the payment channel has a
+single value for m it is computable as, s+(m*i).
+
+- s_e, the _executed iteration sequence number_, is the sequence number that the
+executed iteration e's transaction set starts at. Assuming the history of the
+payment channel has a single value for m it is computable as, s+(m*e).
+
+### Processes
+
+#### Setup
+
+To setup the payment channel:
+
+1. I creates the escrow account E.
+2. R creates the reserve account V.
+3. Set variable initial states:
+   - s to E's sequence number + 1.
+   - i to 0.
+   - e to 0.
+5. I and R build the formation transaction F.
+6. Increment i.
+7. Sign and exchange a closing transaction C_i, that closes the channel with
+disbursements matching the initial contributions.
+8. Sign and exchange a declaration transaction D_i.
+9. I and R sign and exchange signatures for formation transaction F.
+10. I or R submit F.
+
+The transactions are constructed as follows:
+
+- F, the _formation transaction_, deposits I and R's contributions to escrow
+account E, R's reserves to reserve account V, and changes escrow account E and
+reserve account V to be 2-of-2 multisig accounts. F has source account E, and
+sequence number set to s_i.
+
+  F contains operations:
+
+  - Operations sponsored by I:
+    - One `BEGIN_SPONSORING_FUTURE_RESERVES` operation that specifies participant
+    I as a sponsor of future reserves.
+    - One or more `SET_OPTIONS` operations adjusting escrow account E's
+    thresholds such that I and R's signers must both sign, and adding I's
+    signers to E.
+    - One or more `SET_OPTIONS` operations adding I's signers to V.
+    - One or more `CHANGE_TRUST` operations adding trustlines to E.
+    - One `END_SPONSORING_FUTURE_RESERVES` operation that stops I sponsoring
+    future reserves of subsequent operations.
+  - Operations sponsored by R:
+    - One `BEGIN_SPONSORING_FUTURE_RESERVES` operation that specifies reserve
+    account R as a sponsor of future reserves.
+    - One or more `SET_OPTIONS` operations adjusting escrow account V's
+    thresholds such that R and I's signers must both sign, and adding R's
+    signers to V.
+    - One or more `SET_OPTIONS` operations adding R's signers to E.
+    - One `END_SPONSORING_FUTURE_RESERVES` operation that stops R sponsoring
+    future reserves of subsequent operations.
+  - One or more `PAYMENT` operations depositing I's contribution to E.
+  - One or more `PAYMENT` operations depositing R's contribution to E.
+  - One or more `PAYMENT` operations depositing R's reserves to V, for each
+  trustline on E that will be used to sponsor claimable balances at
+  disbursement.
+  
+- C_i, see [Update](#Update) process.
+
+- D_i, see [Update](#Update) process.
+
+#### Update
```

Comment:
A general note about this section: one of the main reasons for updating a payment channel is to effectively make an off-chain payment between the two participants. 

This is clearly enabled by the mechanics described below (if I correctly understood you simply have to update the value on the `CREATE_CLAIMABLE_BALANCE` operation), but maybe it's worth making this more explicit? After all, I would guess this is the main functionality people are looking for when implementing this protocol.

Reply:
Good call. Added in d653b78.

Reply:
Looks great!